User can fork Recipe

shsteindorsson opened the pull request:
User can fork Recipe
================
[PT-link](https://www.pivotaltracker.com/story/show/169667300)
```
As a User
In order to customize an existing recipe
I would like to fork someone's recipe
```
## PR content
In this PR we are adding a feature where a registered user can for a pre-existing recipe so that it can be customized by the user.

## This PR contains:
- Forks controller
- Model method
- Unit test
- route direction to Forks controller

Commit:
Add request spec for forking a recipe

Commit:
Generate Forks controller with create action

Commit:
Add :fork to v1 namespace in router and point to Forks controller

Commit:
Add fork method to recipe model

Add call to fork method to Forks controller.
Update request spec.

Commit:
Add foreign key to recipe model

Add foreign key to recipe model which is the id of the recipe that the
recipe was forked from.
Update request spec.

tochman commented on spec/requests/v1/user_can_fork_recipe_spec.rb:
```diff
@@ -0,0 +1,39 @@
+    create(:recipe,
+           title: 'Cookies',
+           ingredients: 'Cookie ingredients, chocolate chips.',
+           directions: 'Make the cookies.'
```
there's a missing comma. ;-)

Commit:
Add post request (to create recipe) to request spec

Update attribute name in fork method (in recipe model).
Add V1 namespec to Forks controller.

Commit:
Change name of variable to be more descriptive

Co-authored-by: Miyesier miyesier@gmail.com

Commit:
Working request spec for forking a recipe

Co-authored-by: Thomas Ochman <thomas@craftacademy.se>

Commit:
Add association and parent and recipe.
Add response on successful fork.

Co-authored-by: Thomas Ochman <thomas@craftacademy.se>

clalil reviewed (approved):
Approved as long as the conflicting files are resolved.

Commit:
Merge branch 'development' of github.com:CraftAcademy/foodhub-api into fork_recipe
